### PR TITLE
Flux pipeline

### PR DESCRIPTION
**Pull request by sayakpaul**
We are working on uploading the `diffusers` weights to the respective FLUX repositories. Will be done very soon.

### CODE REVIEW

File: src/diffusers/pipelines/flux/pipeline_flux.py [hunk 1]
```diff
@@ -0,0 +1,760 @@
...
+            for i, t in enumerate(timesteps):
+                if self.interrupt:
+                    continue
+
+                # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
+                timestep = t.expand(latents.shape[0]).to(latents.dtype)
+
+                # handle guidance
+                if self.transformer.config.guidance_embeds:
+                    guidance = torch.tensor([guidance_scale])
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/flux/pipeline_flux.py:701]:
```suggestion
                    guidance = torch.tensor([guidance_scale], device=device)
```

**Comment by franz101** [comment]
Amazing job! This model is incredible. Will this allow fine tuning as well?

**Comment by nagolinc** [comment]
when I try the example code

```
import torch
from diffusers import  FluxPipeline

pipe = FluxPipeline.from_pretrained("black-forest-labs/FLUX.1-schnell", torch_dtype=torch.bfloat16)
pipe.enable_model_cpu_offload()

prompt = "A cat holding a sign that says hello world"
out = pipe(
    prompt=prompt, 
    guidance_scale=0., 
    height=768, 
    width=1360, 
    num_inference_steps=4, 
    max_sequence_length=256,
).images[0]
out.save("image.png")
```

I get
`Entry Not Found for url: https://huggingface.co/black-forest-labs/FLUX.1-schnell/resolve/main/model_index.json.`

maybe this file needs to be added to the flux repo?

**Comment by bghira** [comment]
add `revision='refs/pr/3'` to your `from_pretrained` invocation

**Comment by lmxyy** [comment]
What's the revision for `"black-forest-labs/FLUX.1-dev"`?

**Comment by sayakpaul** [comment]
`revision='refs/pr/3'`

**Comment by lmxyy** [comment]
I see. Thanks!

**Comment by loboere** [comment]
add img2img and inpaint pipeline please

**Pull request merged**